superset/models/alerts.py

```diff
@@ -100,3 +105,154 @@ class AlertLog(Model):
+    name = Column(String(150), nullable=False)
+    sql = Column(Text, nullable=False)
+
+    @declared_attr
+    def alert_id(self) -> int:
+        return Column(Integer, ForeignKey("alerts.id"), nullable=False)
+
+    @declared_attr
+    def alert(self) -> RelationshipProperty:
+        return relationship(
+            "Alert",
+            foreign_keys=[self.alert_id],
+            backref=backref("sql_observers", cascade="all, delete-orphan"),
+        )
+
+    @declared_attr
+    def database_id(self) -> int:
+        return Column(Integer, ForeignKey("dbs.id"), nullable=False)
+
+    @declared_attr
+    def database(self) -> RelationshipProperty:
+        return relationship(
+            "Database",
+            foreign_keys=[self.database_id],
+            backref=backref("sql_observers", cascade="all, delete-orphan"),
+        )
+
+    # TODO: Abstract observations from the sqlamodls e.g.
+    # https://github.com/apache/incubator-superset/blob/master/superset/utils/log.py#L32
+    def get_observations(self, observation_num: Optional[int] = 2) -> List[Any]:
+        return (
+            db.session.query(SQLObservation)
+            .filter_by(observer_id=self.id)
+            .order_by(SQLObservation.dttm.desc())
+            .limit(observation_num)
+        )
+
+
+class SQLObservation(Model):  # pylint: disable=too-few-public-methods
+    """Keeps track of values retrieved from SQLObservers"""
+
+    __tablename__ = "sql_observations"
+
+    id = Column(Integer, primary_key=True)
+    dttm = Column(DateTime, default=datetime.utcnow, index=True)
+    observer_id = Column(Integer, ForeignKey("sql_observers.id"), nullable=False)
+    observer = relationship(
+        "SQLObserver",
+        foreign_keys=[observer_id],
+        backref=backref("observations", cascade="all, delete-orphan"),
+    )
+    alert_id = Column(Integer, ForeignKey("alerts.id"))
+    alert = relationship(
+        "Alert",
+        foreign_keys=[alert_id],
+        backref=backref("observations", cascade="all, delete-orphan"),
+    )
+    value = Column(Float)
+
+
+class Validator(Model):
+    """Used to determine how an alert and its observations should be validated"""
+
+    __tablename__ = "alert_validators"
+
+    id = Column(Integer, primary_key=True)
+    name = Column(String(150), nullable=False)
+    validator_type = Column(Enum(AlertValidatorType))
+    config = Column(
+        Text,
+        default=textwrap.dedent(
+            """
+            {
+                "example_threshold": 50
+            }
+            """
+        ),
+    )
+
+    @declared_attr
+    def alert_id(self) -> int:
+        return Column(Integer, ForeignKey("alerts.id"), nullable=False)
+
+    @declared_attr
+    def alert(self) -> RelationshipProperty:
+        return relationship(
+            "Alert",
+            foreign_keys=[self.alert_id],
+            backref=backref("alert_validators", cascade="all, delete-orphan"),
+        )
+
+
+def not_null_executor(
```

Comment:
this should live outside the models, you can create a separate module for them 
e.g. tasks/schedules/alerts/validator.py